aws/resource_aws_waf_geo_match_set.go

```diff
@@ -0,0 +1,201 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/waf"
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsWafGeoMatchSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafGeoMatchSetCreate,
+		Read:   resourceAwsWafGeoMatchSetRead,
+		Update: resourceAwsWafGeoMatchSetUpdate,
+		Delete: resourceAwsWafGeoMatchSetDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"geo_match_constraints": &schema.Schema{
```

Comment:
As discussed in all recent WAF PRs we tend to use singular names for Sets/Lists which have nested fields of non-primitive data types, like here. This is because we expect them to be used in HCL like this:

```hcl
geo_match_constraint {
  type = "Country"
  value = "US"
}
geo_match_constraint {
  type = "Country"
  value = "CA"
}
```
as opposed to lists/sets with primitive data types, like

```hcl
countries = ["CA", "US", "CN"]
```

Do you mind changing it to `geo_match_constraint` and reflecting it in CRUD & docs too?